automation_tools/__init__.py

```diff
@@ -1756,8 +1756,10 @@ def copy_ssh_key(from_host, to_host):
     :param to_host: A string. Hostname on to which the ssh-key will be copied.
 
     """
-    execute(lambda: run('[ ! -f ~/.ssh/id_rsa.pub ] && ssh-keygen '
-                        '-f ~/.ssh/id_rsa -t rsa -N \'\''), host=from_host)
+    execute(lambda: run(
+        '[ ! -f ~/.ssh/id_rsa.pub ] && '
+        'ssh-keygen -f ~/.ssh/id_rsa -t rsa -N \'\'', warn_only=True),
```

Comment:
This should not fail if was not able to generate the key? Adding warn_only will allow the code proceed without this prerequisite.